test/Jamfile.v2

```diff
@@ -47,6 +47,7 @@ test-suite variant
       : variant_noexcept_test
     ]
     [ run recursive_variant_test.cpp : : : <rtti>off <define>BOOST_NO_RTTI <define>BOOST_NO_TYPEID : variant_no_rtti_test ]
+    [ run hash_recursive_variant_test.cpp : : : <cxxflags>-std=c++11 : recursive_variant__hash_test ]
```

Comment:
You do not need the `: : : <cxxflags>-std=c++11 : recursive_variant__hash_test `. Flag `-std=c++11` is not unified across platforms and, for example, the MSVC compiler will fail to compile this test because of the  `-std=c++11`